addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -110,7 +110,7 @@ def _l10n_sa_get_payment_means_code(self, invoice):
     def _get_invoice_payment_means_vals_list(self, invoice):
         """ Override to include/update values specific to ZATCA's UBL 2.1 specs """
         res = super()._get_invoice_payment_means_vals_list(invoice)
-        res[0]['payment_means_code'] = PAYMENT_MEANS_CODE[self._l10n_sa_get_payment_means_code(invoice)]
+        res[0]['payment_means_code'] = PAYMENT_MEANS_CODE.get(self._l10n_sa_get_payment_means_code(invoice)) or PAYMENT_MEANS_CODE['unknown']
```

Comment:
```suggestion
        res[0]['payment_means_code'] = PAYMENT_MEANS_CODE.get(self._l10n_sa_get_payment_means_code(invoice), 1) # or PAYMENT_MEANS_CODE['unknown'] instead of 1  
```